internal/controllers/machineset/machineset_controller.go

```diff
@@ -389,6 +364,43 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 	return ctrl.Result{}, nil
 }
 
+// syncMachines updates Machines to propagate in-place mutable fields from the MachineSet.
+// Note: It also cleans up managed fields of all Machines so that Machines that were created/patched before the
+// controller adopted SSA can also work with SSA.  Otherwise fields would be co-owned by our "old" "manager" and
+// "capi-machineset" and then we would not be able to e.g. drop labels and annotations.
+// TODO: update the labels and annotations to the corresponding infra machines and the boostrap configs of the filtered machines.
+func (r *Reconciler) syncMachines(ctx context.Context, machineSet *clusterv1.MachineSet, machines []*clusterv1.Machine) error {
+	log := ctrl.LoggerFrom(ctx)
+	for i := range machines {
+		m := machines[i]
+		// If the machine is already being deleted, we don't need to update it.
+		if !m.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		// Cleanup managed fields of all Machines.
+		// We do this so that Machines that were created/patched before the controller adopted SSA can also work with SSA.
```

Comment:
could we put a version/timeline on it (e.g. `Machines that were created/patched before the controller adopted SSA (v1.4.0)`)? Down the line, it might be hard to remember which machines are impacted. 

Also, this seems like a workaround to address backwards compatibility, is the plan to remove this code later?

Reply:
> Also, this seems like a workaround to address backwards compatibility, is the plan to remove this code later?

The plan is to remove it eventually, we didn't talk about a timeline yet. I think we should keep it for a few minor releases and then drop it (and track that in an issue)